lib/CodeTransform/Adapter/WorseReflection/Refactor/WorseExpandClass.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Phpactor\CodeTransform\Adapter\WorseReflection\Refactor;
+
+use Microsoft\PhpParser\Node\QualifiedName;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeTransform\Domain\Refactor\ExpandClass;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\Extension\SourceCodeFilesystemExtra\SourceCodeFilestem\Application\ClassSearch;
+use Phpactor\TextDocument\TextDocumentEdits;
+use Phpactor\TextDocument\TextEdit;
+use Phpactor\TextDocument\TextEdits;
+use Phpactor\WorseReflection\Reflector;
+
+class WorseExpandClass implements ExpandClass
+{
+    private Reflector $reflector;
+
+    private ClassSearch $classSearch;
+
+    private Parser $parser;
+
+    public function __construct(
+        Reflector $reflector,
+        ClassSearch $classSearch,
+        Parser $parser = null
+    ) {
+        $this->reflector = $reflector;
+        $this->classSearch = $classSearch;
+        $this->parser = $parser ?: new Parser();
+    }
+
+    public function getTextEdits(SourceCode $sourceCode, int $offset): TextDocumentEdits
+    {
+        $symbolInformation = $this->reflector
+            ->reflectOffset($sourceCode->__toString(), $offset)
+            ->symbolContext()
+            ->symbol();
+
+        $result = $this->classSearch->classSearch('filesystem', $symbolInformation->name());
```

Comment:
no need to use "class search" and that class is very much legacy and should not be used here (note that it doesn't even come from a different "package" but actually from the "application".

you can just:

```
$type = $symbolInformation->type();
if (!$type instanceof ClassType) {
    return empty text edits;
}

$newClassName = $type->name()->__toString();
```

Reply:
Yeah, the new implementation looks much cleaner now as well.